core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -81,7 +81,7 @@
 
   /** Executor for application processing. */
   private Executor executor;
-  @GuardedBy("lock") private boolean usingSharedExecutor;
+  private boolean usingSharedExecutor;
```

Comment:
All the more reason that the synchronization of the variable needs to be clearly specified.  This class is so poorly documented that it requires considerable effort to understand.  

Either add it to the synchronized block, or comment that it is explicitly not synchronized.  As of right now, it looks like it was just forgotten.